Simplified/Catalog/NYPLCatalogUngroupedFeed.m

```diff
@@ -176,11 +184,30 @@ - (void)prepareForBookIndex:(NSUInteger)bookIndex
   
   self.greatestPreparationIndex = bookIndex;
   
-  if(self.currentlyFetchingNextURL) return;
+  if(self.books.count - bookIndex > preloadThreshold) {
+    return;
+  }
   
-  if(!self.nextURL) return;
+  [self fetchNextPageWithCompletionHandler:nil];
+}
+
+// This method might trigger a recursive loop, if and only if
```

Comment:
I would rephrase this a bit because as it is it looks like this recursive loop is not intended, but it is. So it's a little bit confusing. How about 
```
Continues to recursively fetch results from `nextURL` if all the books from last fetch are not supported.
@param handler Called at the end of the recursion.
```
?

(let me know if I'm off, I'm not 100% sure what I wrote is correct)